src/array.rs

```diff
@@ -167,20 +169,23 @@ where
 }
 
 macro_rules! impl_array {
-    ($size:tt) => {
+    ($size:expr) => {
         impl<T, U, F> MapArray<[T; $size], F> for [U; $size]
         where
             F: FnMut(T) -> U,
         {
             fn map_array(mut values: [T; $size], mut f: F) -> Self {
-                use std::{mem::forget, ptr::{read, write}};
+                use std::{
+                    mem::forget,
+                    ptr::{read, write},
+                };
 
                 unsafe {
                     // All elements of `result` is written.
                     // Each element of `values` read once and then forgotten.
                     // Hence safe in case `f` never panics.
                     // TODO: Make it panic-safe.
-                    let mut result: [U; $size] = ::std::mem::uninitialized();
+                    let mut result: [U; $size] = ::std::mem::MaybeUninit::zeroed().assume_init();
```

Comment:
`assume_init()` should be called after all elements are written.
Otherwise it is no better than `uninitialized`